src/spikeinterface/generation/hybrid_tools.py

```diff
@@ -0,0 +1,518 @@
+        if min_amplitude is not None:
+            mask &= amplitudes >= min_amplitude
+        if max_amplitude is not None:
+            mask &= amplitudes <= max_amplitude
+    if min_depth is not None or max_depth is not None:
+        assert templates.probe is not None, "Templates should have a probe to filter based on depth"
+        depth_dimension = ["x", "y"].index(depth_direction)
+        channel_depths = templates.get_channel_locations()[:, depth_dimension]
+        unit_depths = channel_depths[extremum_channel_indices]
+        if min_depth is not None:
+            mask &= unit_depths >= min_depth
+        if max_depth is not None:
+            mask &= unit_depths <= max_depth
+    if np.sum(mask) == 0:
+        warnings.warn("No templates left after filtering")
+        return None
+    filtered_unit_ids = templates.unit_ids[mask]
+    filtered_templates = templates.select_units(filtered_unit_ids)
+
+    return filtered_templates
+
+
+def scale_templates(
+    templates: Templates,
+    min_amplitude: float,
+    max_amplitude: float,
+    amplitude_function: Literal["ptp", "min", "max"] = "ptp",
+):
+    """
+    Scale templates to have a minimum and maximum amplitude.
+
+    Parameters
+    ----------
+    templates : Templates
+        The input templates
+    min_amplitude : float
+        The minimum amplitude of the output templates after scaling
+    max_amplitude : float
+        The maximum amplitude of the output templates after scaling
+
+    Returns
+    -------
+    Templates
+        The scaled templates
+    """
+    extremum_channel_indices = list(get_template_extremum_channel(templates, outputs="index").values())
+    extremum_channel_indices = np.array(extremum_channel_indices, dtype=int)
+
+    # get amplitudes
+    if amplitude_function == "ptp":
+        amp_fun = np.ptp
+    elif amplitude_function == "min":
+        amp_fun = np.min
+    elif amplitude_function == "max":
+        amp_fun = np.max
+    amplitudes = np.zeros(templates.num_units)
+    templates_array = templates.templates_array
+    for i in range(templates.num_units):
+        amplitudes[i] = amp_fun(templates_array[i, :, extremum_channel_indices[i]])
+
+    # scale templates to meet min_amplitude and max_amplitude range
+    min_scale = np.min(amplitudes) / min_amplitude
+    max_scale = np.max(amplitudes) / max_amplitude
+    m = (max_scale - min_scale) / (np.max(amplitudes) - np.min(amplitudes))
+    scales = m * (amplitudes - np.min(amplitudes)) + min_scale
+
+    scaled_templates_array = templates.templates_array / scales[:, None, None]
+
+    return Templates(
+        templates_array=scaled_templates_array,
+        sampling_frequency=templates.sampling_frequency,
+        nbefore=templates.nbefore,
+        sparsity_mask=templates.sparsity_mask,
+        channel_ids=templates.channel_ids,
+        unit_ids=templates.unit_ids,
+        probe=templates.probe,
+    )
+
+
+def shift_templates(
```

Comment:
shift is a bit fuzy no ? For me shift is more somethign shift in time.
Why not move_templates() ?

Reply:
And also I would call it random_move_templates() this is more explcit.

Reply:
`relocate_templates`